renderdoc/driver/vulkan/vk_overlay.cpp

```diff
@@ -1587,7 +1587,10 @@ ResourceId VulkanReplay::RenderOverlay(ResourceId texid, CompType typeHint, Debu
       {
         blackclear.colorAttachment =
             rp.subpasses[m_pDriver->m_RenderState.subpass].colorAttachments[i];
-        atts.push_back(blackclear);
+        if(blackclear.colorAttachment != VK_ATTACHMENT_UNUSED)
+        {
+          atts.push_back(blackclear);
+        }
```

Comment:
Sorry I should have done this last night, but I looked at the docs for `vkCmdClearAttachments` and for `VkClearAttachment::colorAttachment` the spec says:

> `colorAttachment` is only meaningful if `VK_IMAGE_ASPECT_COLOR_BIT` is set in `aspectMask`, in which case it is an index to the `pColorAttachments` array in the VkSubpassDescription structure of the current subpass which selects the color attachment to clear.

It also says in `vkCmdClearAttachments`:

> `pAttachments` is a pointer to an array of `VkClearAttachment` structures defining the attachments to clear and the clear values to use. **If any attachment to be cleared in the current subpass is `VK_ATTACHMENT_UNUSED`, then the clear has no effect on that attachment.**

So maybe this indirection and loop is all wrong to start with, and it should just be `blackclear.colorAttachment = i;` effectively? Rather than indices into the framebuffer attachment list as it is now.

I have few renderpass tests so I'd guess when I've tested this code it's always been with a simple case of subpass colorattachment indices being identical to framebuffer indices. I could construct a test case where that isn't true to test with.

Reply:
Indeed, on second look - `i` is already an index into color attachments in subpass.

Reply:
Huh, not that easy, `blackclear.colorAttachment = i;` doesn't work (nothing is cleared) so looking into it.

Reply:
Sorry, I managed to make a mistake in one line... Obviously it is working.

Reply:
Great, thanks!

Reply:
There's an issue with the appveyor build - you need an explicit narrowing cast like `blackclear.colorAttachment = (uint32_t)i;` for 64-bit windows, as `i` is `size_t` which is 64-bit.

I'll merge once CI is green.